lib/manageiq_performance/middlewares/memory.rb

```diff
@@ -0,0 +1,58 @@
+module ManageIQPerformance
+  module Middlewares
+    module Memory
+      private
+
+      def memory_initialize
+      end
+
+      def memory_start(env)
+        self.memory_query_data = memory_parse_data
+      end
+
+      def memory_finish(env)
+        self.memory_query_data = memory_parse_data(memory_query_data)
+        save_report env, :memory,
+                    memory_short_form_data,
+                    memory_long_form_data
+      ensure
+        self.memory_query_data = nil
+      end
+
+      def memory_long_form_data
+        proc { memory_query_data }
+      end
+
+      def memory_short_form_data
+        proc { Memory.memory_query_data }
```

Comment:
I think this one is incorrect in it's usage (since `memory_query_data` is a instance method now), and should just be the same as `memory_long_form_data` (otherwise that one is incorrect, though, I doubt it).